sw/ground_segment/python/natnet3.x/natnet2ivy.py

```diff
@@ -142,22 +159,57 @@ def receiveRigidBodyList( rigidBodyList, stamp ):
             continue # too early for next message
         timestamp[i] = stamp
 
-        msg = PprzMessage("datalink", "REMOTE_GPS_LOCAL")
-        msg['ac_id'] = id_dict[i]
-        msg['pad'] = 0
-        msg['enu_x'] = pos[0]
-        msg['enu_y'] = pos[1]
-        msg['enu_z'] = pos[2]
         vel = compute_velocity(i)
-        msg['enu_xd'] = vel[0]
-        msg['enu_yd'] = vel[1]
-        msg['enu_zd'] = vel[2]
-        msg['tow'] = int(1000. * stamp) # TODO convert to GPS itow ?
-        # convert quaternion to psi euler angle
-        dcm_0_0 = 1.0 - 2.0 * (quat[1] * quat[1] + quat[2] * quat[2])
-        dcm_1_0 = 2.0 * (quat[0] * quat[1] - quat[3] * quat[2])
-        msg['course'] = 180. * np.arctan2(dcm_1_0, dcm_0_0) / 3.14
-        ivy.send(msg)
+
+        # Rotate everything if Z-forward
+        if args.z_forward:
+            pos = rotZtoX(pos)
+            vel = rotZtoX(vel)
+            quat = rotZtoX(quat, True)
+
+        # Check which message to send
+        if args.rgl_msg:
+            msg = PprzMessage("datalink", "REMOTE_GPS_LOCAL")
+            msg['ac_id'] = id_dict[i]
+            msg['pad'] = 0
+            msg['enu_x'] = pos[0]
+            msg['enu_y'] = pos[1]
+            msg['enu_z'] = pos[2]
+            msg['enu_xd'] = vel[0]
+            msg['enu_yd'] = vel[1]
+            msg['enu_zd'] = vel[2]
+            msg['tow'] = int(1000. * stamp) # TODO convert to GPS itow ?
+            # convert quaternion to psi euler angle
+            dcm_0_0 = 1.0 - 2.0 * (quat[1] * quat[1] + quat[2] * quat[2])
+            dcm_1_0 = 2.0 * (quat[0] * quat[1] - quat[3] * quat[2])
+            msg['course'] = 180. * np.arctan2(dcm_1_0, dcm_0_0) / 3.14
+            ivy.send(msg)
+        elif args.small_msg:
+            # First check if position fits in message
+            if abs(pos[0]*100.0) > pow(2, 10) or abs(pos[1]*100.0) > pow(2, 10) or pos[2]*100.0 > pow(2, 10) or pos[2] < 0.:
+                print("Position out of range for small message")
+                continue
+
+            # TODO calculate everything
+            msg = PprzMessage("datalink", "EXTERNAL_POSE_SMALL")
+            msg['ac_id'] = id_dict[i]
+            msg['timestamp'] = int(1000. * stamp) # Time in ms
+            ivy.send(msg)
+        else:
+            msg = PprzMessage("datalink", "EXTERNAL_POSE")
+            msg['ac_id'] = id_dict[i]
+            msg['timestamp'] = int(1000. * stamp) # Time in ms
+            msg['enu_x'] = pos[0]
+            msg['enu_y'] = pos[1]
+            msg['enu_z'] = pos[2]
+            msg['enu_xd'] = vel[0]
+            msg['enu_yd'] = vel[1]
+            msg['enu_zd'] = vel[2]
+            msg['body_qi'] = quat[3]
```

Comment:
The quaternion is representing the transformation in ENU frame. This is a problem when reconstruction the heading onboard. We have to decide the proper format.